bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/api/DeviceController.java

```diff
@@ -0,0 +1,261 @@
+        String response = apiGetMethod(makeURL("api/v1/inverter/" + this.sn + "/realtime/input"),
+                APIdata.staticAccessToken);
+        Gson gson = new Gson();
+        @Nullable
+        RealTimeInData realTimeInData = gson.fromJson(response, RealTimeInData.class);
+        if (realTimeInData == null)
+            throw new SunSynkDeviceControllerException("Could not retrieve solar state");
+        this.realTimeDataIn = realTimeInData;
+        this.realTimeDataIn.sumPVIV();
+    }
+
+    /**
+     * Sends the internal battery charge and discharge settings to the concrete device (inverter)
+     * of a Sun Synk Connect Account
+     * 
+     * @param settings
+     * @throws SunSynkSendCommandException
+     */
+    public void sendSettings(Settings settings) throws SunSynkSendCommandException {
+        String body = settings.buildBody();
+        sendCommandToSunSynk(body);
+    }
+
+    private void sendCommandToSunSynk(String body) throws SunSynkSendCommandException {
+        String path = "api/v1/common/setting/" + this.sn + "/set";
+
+        try {
+            apiPostMethod(makeURL(path), body, APIdata.staticAccessToken);
+        } catch (IOException e) {
+            logger.debug("Failed to send to Inverter API: {} ", e.getMessage());
+            int found = e.getMessage().indexOf("Authentication challenge without WWW-Authenticate header");
+            if (found > -1) {
+                throw new SunSynkSendCommandException("Authentication token failed", e);
+            }
+            throw new SunSynkSendCommandException("Unknown athentication fail", e);
+        }
+        logger.debug("Sent command: to inverter {}.", this.sn);
+    }
+
+    private String apiPostMethod(String httpsURL, String body, String accessToken) throws IOException {
+        Properties headers = new Properties();
+        headers.setProperty("Accept", "application/json");
+        headers.setProperty("Authorization", "Bearer " + accessToken);
+        InputStream stream = new ByteArrayInputStream(body.getBytes(StandardCharsets.UTF_8));
+        return HttpUtil.executeUrl(HttpMethod.POST.asString(), httpsURL, headers, stream, "application/json",
+                TIMEOUT_IN_MS);
+    }
+
+    private String apiGetMethod(String httpsURL, String accessToken) throws IOException {
+        Properties headers = new Properties();
+        headers.setProperty("Accept", "application/json");
+        headers.setProperty("Authorization", "Bearer " + accessToken);
+        return HttpUtil.executeUrl(HttpMethod.GET.asString(), httpsURL, headers, null, "application/json",
+                TIMEOUT_IN_MS);
+    }
+
+    private String makeURL(String path) {
+        return "https://api.sunsynk.net" + "/" + path;
+    }
+
+    private String getAPIFormatDate() {
+        LocalDate date = LocalDate.now();
+        DateTimeFormatter APIformat = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        return date.format(APIformat);
```

Comment:
SAT local var name. But even better to change it to a one liner:
```suggestion
        return LocalDate.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
```

Reply:
ok